executor/write_test.go

```diff
@@ -1403,6 +1480,40 @@ func (s *testSuite8) TestUpdate(c *C) {
 	tk.MustExec("create table t1(a int, b int, c int, d int, e int, index idx(a))")
 	tk.MustExec("create table t2(a int, b int, c int)")
 	tk.MustExec("update t1 join t2 on t1.a=t2.a set t1.a=1 where t2.b=1 and t2.c=2")
+
+	// Assign `DEFAULT` in `UPDATE` statement
+	tk.MustExec("drop table if exists t1, t2;")
+	tk.MustExec("create table t1 (a int default 1, b int default 2);")
+	tk.MustExec("insert into t1 values (10, 10), (20, 20);")
+	tk.MustExec("update t1 set a=default where b=10;")
+	tk.MustQuery("select * from t1;").Check(testkit.Rows("1 10", "20 20"))
+	tk.MustExec("update t1 set a=30, b=default where a=20;")
+	tk.MustQuery("select * from t1;").Check(testkit.Rows("1 10", "30 2"))
+	tk.MustExec("update t1 set a=default, b=default where a=30;")
+	tk.MustQuery("select * from t1;").Check(testkit.Rows("1 10", "1 2"))
+	tk.MustExec("insert into t1 values (40, 40)")
+	tk.MustExec("update t1 set a=default, b=default")
+	tk.MustQuery("select * from t1;").Check(testkit.Rows("1 2", "1 2", "1 2"))
+	tk.MustExec("update t1 set a=default(b), b=default(a)")
+	tk.MustQuery("select * from t1;").Check(testkit.Rows("2 1", "2 1", "2 1"))
+	// With generated columns
+	tk.MustExec("create table t2 (a int default 1, b int generated always as (-a) virtual, c int generated always as (-a) stored);")
+	tk.MustExec("insert into t2 values (10, default, default), (20, default, default)")
+	tk.MustExec("update t2 set b=default;")
+	tk.MustQuery("select * from t2;").Check(testkit.Rows("10 -10 -10", "20 -20 -20"))
+	tk.MustExec("update t2 set a=30, b=default where a=10;")
+	tk.MustQuery("select * from t2;").Check(testkit.Rows("30 -30 -30", "20 -20 -20"))
+	tk.MustExec("update t2 set c=default, a=40 where c=-20;")
+	tk.MustQuery("select * from t2;").Check(testkit.Rows("30 -30 -30", "40 -40 -40"))
+	tk.MustExec("update t2 set a=default, b=default, c=default where b=-30;")
+	tk.MustQuery("select * from t2;").Check(testkit.Rows("1 -1 -1", "40 -40 -40"))
+	tk.MustExec("update t2 set a=default(a), b=default, c=default;")
+	tk.MustQuery("select * from t2;").Check(testkit.Rows("1 -1 -1", "1 -1 -1"))
+	tk.MustGetErrCode("update t2 set b=default(a);", mysql.ErrBadGeneratedColumn)
```

Comment:
What about test for `update t2 set b=default(b);`? Seems that mysql allows it.

Reply:
@lamxTyler  We have discussed this case, and have agreement that only allow `default` keyword for generated column assignment. Use `default(a)` in here is disallowed. e.g.
```sql
create table t (a int, b int as (-a));
insert into t value (1, default);
update t set b = default;    # both allowed in MySQL and TiDB
update t set b = default(a); # allowed in MySQL, but disallowed in TiDB
update t set b = default(b); # allowed in MySQL, but disallowed in TiDB
```

Reply:
Can you comment the reason that only allow `default` in code?